### PR TITLE
Allow passing schemaUrl to tracer and meter provider, separate sdk options to a new class

### LINKED ISSUE #41485: OTel tracing: add option to provide lib-specific schema URL
`TracingOptions` are user config, but we want libraries (like cosmos) to provide version of otel schema they emit. Users should not control it.

We use static version, but Cosmos and EventHubs/SB may emit different versions and we can't keep them in sync.

```java
public Tracer createTracer(LibraryTracingOptions libOptions, TracingOptions options)
```

(same on metrics)

### PR DESCRIPTION
**Pull request by lmolkova**
Fix #41485

### CODE REVIEW

File: sdk/core/azure-core-metrics-opentelemetry/src/samples/java/com/azure/core/metrics/opentelemetry/MetricsJavaDocCodeSnippets.java [hunk 1]
```diff
@@ -192,7 +193,12 @@ private static final class AzureClient {
         private final DoubleHistogram callDuration;
         private final TelemetryAttributes attributes;
         AzureClient(String endpoint, ClientOptions options) {
-            meter = DEFAULT_PROVIDER.createMeter("azure-core-samples", "1.0.0", options == null ? null : options.getMetricsOptions());
+            SdkTelemetryOptions sdkTelemetryOptions = new SdkTelemetryOptions()
+                .setSdkName("azure-samples")
+                .setSdkVersion("1.0.0")
+                .setSdkVersion("Microsoft.Sample")
```
Comment (trask) [inline sdk/core/azure-core-metrics-opentelemetry/src/samples/java/com/azure/core/metrics/opentelemetry/MetricsJavaDocCodeSnippets.java:199]:
```suggestion
                .setResourceProviderNamespace("Microsoft.Sample")
```

File: sdk/core/azure-core-tracing-opentelemetry-samples/src/samples/java/com/azure/core/tracing/opentelemetry/samples/TracingJavaDocCodeSnippets.java [hunk 2]
```diff
@@ -142,8 +143,13 @@ private static final class AzureClient {
         private static final TracerProvider DEFAULT_PROVIDER = TracerProvider.getDefaultProvider();
         private final com.azure.core.util.tracing.Tracer tracer;
         AzureClient(String endpoint, ClientOptions options) {
-            tracer = DEFAULT_PROVIDER.createTracer("azure-storage-blob", "12.20.0",
-                "Microsoft.Storage", options == null ? null : options.getTracingOptions());
+            SdkTelemetryOptions sdkTelemetryOptions = new SdkTelemetryOptions()
+                .setSdkName("azure-sample")
+                .setSdkVersion("12.20.0")
+                .setSdkVersion("Microsoft.Sample")
```
Comment (trask) [inline sdk/core/azure-core-tracing-opentelemetry-samples/src/samples/java/com/azure/core/tracing/opentelemetry/samples/TracingJavaDocCodeSnippets.java:149]:
```suggestion
                .setResourceProviderNamespace("Microsoft.Sample")
```

File: sdk/core/azure-core-metrics-opentelemetry/src/main/java/com/azure/core/metrics/opentelemetry/OpenTelemetryMeter.java [hunk 3]
```diff
@@ -28,15 +29,21 @@ class OpenTelemetryMeter implements Meter {
...
         }
 
-        this.isEnabled = (options == null || options.isEnabled()) && otelProvider != MeterProvider.noop();
-        this.meter = otelProvider.meterBuilder(libraryName).setInstrumentationVersion(libraryVersion).build();
+        this.isEnabled
+            = (applicationOptions == null || applicationOptions.isEnabled()) && otelProvider != MeterProvider.noop();
+        this.meter = otelProvider.meterBuilder(sdkOptions.getSdkName())
+            .setInstrumentationVersion(sdkOptions.getSdkVersion())
+            .setSchemaUrl(sdkOptions.getSchemaUrl())
+            .build();
```
Comment (trask) [inline sdk/core/azure-core-metrics-opentelemetry/src/main/java/com/azure/core/metrics/opentelemetry/OpenTelemetryMeter.java:46]: just checking that intentionally not using `resourceProviderNamespace` here?

**Review by trask — APPROVED** [review]

File: sdk/core/azure-core-metrics-opentelemetry/src/samples/java/com/azure/core/metrics/opentelemetry/MetricsJavaDocCodeSnippets.java (hunk 1 above)
Comment (lmolkova) [reply sdk/core/azure-core-metrics-opentelemetry/src/samples/java/com/azure/core/metrics/opentelemetry/MetricsJavaDocCodeSnippets.java]: good catch, thanks!

File: sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracerProvider.java [hunk 4]
```diff
@@ -29,12 +30,29 @@ public OpenTelemetryTracerProvider() {
...
+    /**
+     * Creates OpenTelemetry-based implementation of {@link Tracer}
+     *
+     * @param libraryOptions Library-specific telemetry options.
+     * @param applicationOptions Tracing options configured by the application.
+     * @return a tracer instance.
+     */
+    @Override
+    public Tracer createTracer(LibraryTelemetryOptions libraryOptions, TracingOptions applicationOptions) {
+        return new OpenTelemetryTracer(libraryOptions, applicationOptions);
```
Comment (alzimmermsft) [inline sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracerProvider.java:56]: Other provider checked for `libraryOptions != null`, should this do the same?

File: sdk/core/azure-core/src/main/java/com/azure/core/util/LibraryTelemetryOptions.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.tracing.TracerProvider;
+
+/**
+ * The options to configure library-specific information on {@link TracerProvider}
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/LibraryTelemetryOptions.java:10]: Isn't this also used with `MeterProvider`?

**Review by alzimmermsft — APPROVED** [review]

File: sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracerProvider.java [hunk 6]
```diff
@@ -29,12 +30,32 @@ public OpenTelemetryTracerProvider() {
...
+     * Creates OpenTelemetry-based implementation of {@link Tracer}
+     *
+     * @param libraryOptions Library-specific telemetry options.
+     * @param applicationOptions Tracing options configured by the application.
+     * @return a tracer instance.
+     */
+    @Override
+    public Tracer createTracer(LibraryTelemetryOptions libraryOptions, TracingOptions applicationOptions) {
+        Objects.requireNonNull(libraryOptions, "'libraryOptions' cannot be null.");
+        Objects.requireNonNull(libraryOptions.getLibraryName(), "'libraryOptions.getLibraryName()' cannot be null.");
```
Comment (srnagar) [inline sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracerProvider.java:57]: Is `libraryName` a required param for all? If so, we should make this is a constructor param in `LibraryTelemetryOptions`

**Review by srnagar — APPROVED** [review]

File: sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracerProvider.java (hunk 6 above)
Comment (lmolkova) [reply sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracerProvider.java]: right!

**Pull request merged**